File: src/main/java/es.weso.shexl/ast/ASTNode.java
package es.weso.shexl.ast;

public interface ASTNode {

    /** The Constant DEFAULT_ROW_COLUMN. */
    int DEFAULT_ROW_COLUMN = -1;

    /**
     * Gets the line where the element that is contained by the node is written.
     *
     * @return the line where the element that the node contains is written in
     *         the input file.
     */
    int getLine();

    /**
     * Gets the column where the element that is contained by the node is
     * written.
     *
     * @return the column where the element that the node contains is written in
     *         the input file.
     */
    int getColumn();
}

File: src/es/weso/shexl/compiler/parser/ast/ASTNode.scala
package es.weso.shexl.compiler.parser.ast

/**
 * AST node represents a node in the Abstract Syntax Tree (AST). The nodes are generated
 * by the parser. Later they are used at semantic analysis and the tree will be attributed
 * with other information that will transform the tree in a graph but the nodes will be
 * the same but with more information. Each node always contains the file, line and column
 * where it was generated. So in case of an error we can trace back to the source code.
 */
private[compiler] trait ASTNode {

  /**
   * Gets the name of the file where the node was originated.
   *
   * @return the name of the file where the node was originated.
   */
  def getFileName: String

  /**
   * Gets the number of the line where the node was originated.
   *
   * @return the number of the line where the node was originated.
   */
  def getLine: Int

  /**
   * Gets the number of the column where the node was originated.
   *
   * @return the number of the column where the node was originated.
   */
  def getColumn: Int
}
